week-four/app/src/main/java/pro/marvinhosea/movielist/repository/UserSharedPrefRepository.kt

```diff
@@ -0,0 +1,72 @@
+package pro.marvinhosea.movielist.repository
+
+import android.content.Context
+import androidx.preference.PreferenceManager
+
+object UserSharedPrefRepository : UserRepository {
+
+    private lateinit var applicationContext: Context
+    private const val USERNAME = "USERNAME"
+    private const val USER_PASSWORD = "USER_PASSWORD"
+    private const val USER_IS_LOGGED_IN = "USER_IS_LOGGED_IN"
+
+    private fun sharedPreps() = PreferenceManager.getDefaultSharedPreferences(applicationContext)
+
+    fun init(context: Context) {
+        applicationContext = context
+    }
+
+    /**
+     * Register user on shared preference
+     */
+    override fun registerUser(username: String, password: String) {
+        val editor = sharedPreps().edit()
+
+        editor.putString(USERNAME, username).apply()
```

Comment:
You should add all the changes first, then call apply.

Because if you have multiple apply calls on the same editor, only the last one will be saved. You have a race condition here basically.